Sources/Datadog/DatadogInternal/DatadogCoreProtocol.swift

```diff
@@ -19,10 +19,13 @@ internal struct FeatureStorageConfiguration {
     /// Each path is relative to the root folder of given SDK instance.
     struct Directories {
         /// The path for writing authorized data (when tracking consent is granted).
+        /// This path must be relative to the core directory created for given instance of the SDK.
         let authorized: String
         /// The path for writing unauthorized data (when tracking consent is pending).
+        /// This path must be relative to the core directory created for given instance of the SDK.
         let unauthorized: String
         /// The list of deprecated paths from previous versions of this feature. It is used to perform cleanup.
+        /// These paths must be relative to the known OS location (`/Library/Caches/`) containing core directories for different instances of the SDK.
         let deprecated: [String]
```

Comment:
> a vendor SDK installed in an app where both use Datadog SDK but vendor uses V2 and app is on V1

If both versions mismatch, the app won't be able to install the vendor framework as a dependency, except if the vendor is manually linking the Datadog SDK as a static lib, which should be discouraged IMO. The app would have to upgrade to v2 or use a previous version of the vendor SDK to match the version. So I wouldn't consider this use case.

Now, multiple instances of the v2 SDK will perform the same operation on the `deprecated` v1 directories, so I wonder if this is the right place for v1 depreciation 🤔 We can consider this as a temporary measure and eventually remove this `deprecated` when we consider good adoption for v2 (which can take time..)

Reply:
👍 Right, I think it's fair to assume there will be a single version of the SDK even if multiple instances are enabled.

Indeed, there's a room for improvement around deprecation logic. Because it requires more refactoring in different parts of the codebase, I created `RUMM-2312` to focus solely on data deprecation logic. For now, I'll keep it as it is - capable of removing global SDK folders, without ensuring the logic is ran only once.